qcengine/programs/openmm.py

```diff
@@ -204,9 +205,9 @@ def compute(self, input_data: "AtomicInput", config: "TaskConfig") -> "AtomicRes
             state = context.getState(getEnergy=True)
 
             # Get the potential as a simtk.unit.Quantity, put into units of hartree
-            q = state.getPotentialEnergy() / unit.hartree
```

Comment:
We use the programs conversion factors so that the values are always the same. I would suggest keeping this here as everyones kJ is slightly different as it rarely follows canonical CODATA.

Reply:
That makes sense but the return value from this is not what we expect. I think energies should be in hartrees but from my example, in issue #224 you can see that they are not. Should I raise this on the openmm github instead?

Reply:
I am not familiar enough with the OpenMM unit system to say, but I think you will need to divide my Avagodros number or similar here to make the units work out. @jchodera @j-wags

Reply:
Yes, I believe that is what the difference is in this case:
```python
# get in hartrees using simtk.unit.AVOGADRO_CONSTANT_NA
result.return_result / unit.AVOGADRO_CONSTANT_NA
```
> -0.05675498965565169

compared to the value using qcelemental
> -0.05602423977045684

Reply:
See my comments on the issue for what I understand is happening in the unit system:

https://github.com/MolSSI/QCEngine/issues/224#issuecomment-581959798

Reply:
Wow! This is why we _always_ want to use internal units. It should be much more consistent.

Reply:
If I recall correctly (and if I understand what you're attempting to do), the internal units for OpenMM is in a [`kJ/mol` basis](http://docs.openmm.org/latest/userguide/theory.html#units), so internally, the double returned by `state.getPotentialEnergy()` is the internal one, just wrapped in a `Unit` class, so if you wanted to use a different unit conversion kit, that should still give you as close to the correct internal units as possible.

Reply:
If that's correct and allows us to use the qcelemental conversion method I would vote for that as I feel it is much cleaner to implement and understand whats going on.

Reply:
I don't know all of the details of OpenMM internals, so a bit of caution here. I know it does a conversion to the internal unit system which I linked above, and decently sure it returns that as well unmodified. 

If you want just the raw number out of a `Unit` class, you can access it through one of two means (assume the object is `X`)

* X._value
* X.value_in_unit(x.unit)

The former just spits out the numeric float attached to the unit, but has the problem of accessing a non-public method, the later uses only public methods, but might result in a small adjustment of the value. I don't know if there is a special check for "if the same unit" which bypasses multiplying by the `simtk.unit` conversion factors. However, since the base units of energy and amount are kJ and mole respectively, I think their conversion factor is literally 1, so its doing `X * 1 / 1` internally (I think), just returning the double anyways.

If it was anything other than the internal units returned by `getPotentialEnergy`, I would be worried about the 2nd approach.

You should not have to worry about `getPotentialEnergy`'s unit returns changing. The internal OpenMM unit system has been the same for a long time and I don't think there is any appetite to do something different in the future. At least, no one has suggested changing it, ever.

Reply:
Thanks for the detailed explanation of how the units are working, I did a quick test and it looks like requesting the energy explicitly in KJ/mol gives the same value as `q._value` (option 1 which is what I have used in this PR):

```python
# now try using the in units of method
q = state.getPotentialEnergy() 
q.value_in_unit(q.unit)
```
> -149.01023036563316

```python
# compare with the _value
q = state.getPotentialEnergy() 
print(q._value)
```
> -149.01023036563316

Reply:
Sure, but I think the only reason it works is because the base conversion the `unit` module is doing is `* 1 / 1`. Personally, I would assume that the units coming out of the `getPotentialEnergy` function are always kJ/mol (a safe assumption), and just use `._value` to avoid any and all possibility of some oddity with the `unit` module I don't know, then use QCElemental's CODATA to do all the conversions. 

It would behoove you to make a few detailed comments about what you are doing there since its a private method access and makes a few assumptions that won't be obvious to other Engine devs/users who look at the code and not this issue. Try and minimize future development issues going forward?

Reply:
The error here is 0.5 kcal/mol between QCElemental/OpenMM units which is pretty scary according to the first numbers. Can we track this down?

Test failures are from another source and not you.

Reply:
Ahhh my mistake I think I have posted the energy using the qcelemental conversion for a different conformer, I have recalculated it for the same conformer and get 
> -0.05675499938859474 for `qcelemental`

and 

> -0.05675498965565169 for `simtk.unit`

which matches what I posted above and is much closer :+1:

Reply:
Thanks for following up on this. I was pretty concerned about the original numbers, but this looks fine!

Reply:
That makes me feel much better too!